core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -102,6 +106,26 @@ case class Uri(
   override protected def self: Self = this
 
   override protected def replaceQuery(query: Query): Self = copy(query = query)
+
+  /**
+   * Converts a [[Uri]] into an [[InetSocketAddress]] as long as this [[Uri]] has an authority.
+   * This conversion is a pretty straightforward translation, most of the work is inserting default port values
+   * if the [[Uri]] does not specify explicit ports.
+   * This method only has knowledge of the Http protocol (including websockets), it does not know how to guess ports
+   * for schemes it has no knowledge of and will return an [[IOException]] if it encounters an unknown scheme.
+   * If the uri does not explicit a scheme, it will assume the [[Http]] scheme.
+   */
+  def asAddress: Throwable \/ InetSocketAddress = {
```

Comment:
Thx for spotting that! I had originally considered the explicit port, but then refactored the implementation to "improve" legibility but forgot to run the tests again.

I tried my hand again at writing an easy implementation to follow, ran the tests this time, hopefully you like it!